src/box-node-sdk.ts

```diff
@@ -0,0 +1,463 @@
+	static collaborationRoles = BoxSDKNode.prototype.collaborationRoles;
+	static CURRENT_USER_ID = BoxSDKNode.prototype.CURRENT_USER_ID;
+
+	/**
+	 * Expose Webhooks.validateMessage() to the SDK as a whole. This allows
+	 * the consumer to call BoxSDK.validateWebhookMessage() by just requiring the SDK,
+	 * instead of needing to create a client (which is not needed to validate messages).
+	 */
+	static validateWebhookMessage = Webhooks.validateMessage;
+
+	constructor(params: UserConfigurationOptions) {
+		super();
+
+		var eventBus = new EventEmitter();
+
+		var self = this;
+		eventBus.on('response', function () {
+			var args: any /* FIXME */ = [].slice.call(arguments);
+			args.unshift('response');
+			self.emit.apply(self, args);
+		});
+
+		// Setup the configuration with the given params
+		this.config = new Config(params);
+		this._eventBus = eventBus;
+		this._setup();
+	}
+
+	/**
+	 * Setup the SDK instance by instantiating necessary objects with current
+	 * configuration values.
+	 *
+	 * @param {EventEmitter} eventBus The event bus to use
+	 * @returns {void}
+	 * @private
+	 */
+	_setup() {
+		// Instantiate the request manager
+		this.requestManager = new APIRequestManager(this.config, this._eventBus);
+
+		// Initialize the rest of the SDK with the given configuration
+		this.tokenManager = new TokenManager(this.config, this.requestManager);
+		this.anonymousSession = new AnonymousAPISession(
+			this.config,
+			this.tokenManager
+		);
+	}
+
+	/**
+	 * Gets the BoxSDKNode instance by passing boxAppSettings json downloaded from the developer console.
+	 *
+	 * @param {Object} appConfig boxAppSettings object retrieved from Dev Console.
+	 * @returns {BoxSDKNode} an instance that has been preconfigured with the values from the Dev Console
+	 */
+	static getPreconfiguredInstance(appConfig: any /* FIXME */) {
+		if (typeof appConfig.boxAppSettings !== 'object') {
+			throw new TypeError(
+				'Configuration does not include boxAppSettings object.'
+			);
+		}
+
+		var boxAppSettings = appConfig.boxAppSettings;
+		var webhooks = appConfig.webhooks;
+		if (typeof webhooks === 'object') {
+			Webhooks.setSignatureKeys(webhooks.primaryKey, webhooks.secondaryKey);
+		}
+
+		var params: {
+			clientID?: string;
+			clientSecret?: string;
+			appAuth?: {
+				keyID?: string;
+				privateKey?: string;
+				passphrase?: string;
+			};
+			enterpriseID?: string;
+		} = {};
+
+		if (typeof boxAppSettings.clientID === 'string') {
+			params.clientID = boxAppSettings.clientID;
+		}
+
+		if (typeof boxAppSettings.clientSecret === 'string') {
+			params.clientSecret = boxAppSettings.clientSecret;
+		}
+
+		// Only try to assign app auth settings if they are present
+		// Some configurations do not include them (but might include other info, e.g. webhooks)
+		if (
+			typeof boxAppSettings.appAuth === 'object' &&
+			boxAppSettings.appAuth.publicKeyID
+		) {
+			var appAuth: Exclude<typeof params['appAuth'], void> = {};
```

Comment:
Slight preference for `params.appAuth` syntax, to be consistent with the rest of the file